web_poet/serialization/functions.py

```diff
@@ -115,9 +118,16 @@ def _serialize_HttpClient(o: HttpClient) -> SerializedLeafData:
         serialized_request = serialize_leaf(data.request)
         for k, v in serialized_request.items():
             serialized_data[f"{i}-HttpRequest.{k}"] = v
-        serialized_response = serialize_leaf(data.response)
-        for k, v in serialized_response.items():
-            serialized_data[f"{i}-HttpResponse.{k}"] = v
+        if data.response:
+            serialized_response = serialize_leaf(data.response)
+            for k, v in serialized_response.items():
+                serialized_data[f"{i}-HttpResponse.{k}"] = v
+        if data.exception:
```

Comment:
Would this be right?

```suggestion
        else:
            assert data.exception
```

Reply:
Well, assuming the data on-disk is well-formed.